src/main/java/graphql/servlet/AbstractGraphQLHttpServlet.java

```diff
@@ -134,7 +150,9 @@ public void init() {
 
                         String operationName = request.getParameter("operationName");
 
-                        query(queryInvoker, graphQLObjectMapper, invocationInputFactory.createReadOnly(new GraphQLRequest(query, variables, operationName), request, response), response);
+                        query(queryInvoker, graphQLObjectMapper,
+                            invocationInputFactory.createReadOnly(new GraphQLRequest(query, variables, operationName), request, response),
+                            Optional.of(request), response);
```

Comment:
It's recommended not to use `Optional` as parameter, see https://rules.sonarsource.com/java/RSPEC-3553